.github/workflows/ci.yml

```diff
@@ -12,17 +12,49 @@ env:
   POETRY_VIRTUALENVS_CREATE: false
 
 jobs:
-  test-build-docs:
+  test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+          cache-dependency-path: "poetry.lock"
+
       - name: Install poetry
         run: pip install poetry
-      - uses: actions/setup-python@v3
+
+      - name: Install dependencies
+        run: poetry install
+
+      - name: Lint
+        run: make lint
+
+      - name: Run docker-compose stack for testing
+        run: make docker-up
+
+      - name: Test
+        # Wait for RabbitMQ to start.
+        run: sleep 10 && make test
```

Comment:
I don't like this; it would be nice for the tests to wait for RabbitMQ to actually be ready. Also, for some reason the tests didn't do anything for 10 minutes on 3 runs, so maybe this should be disabled?

Suggestions welcome.